api/views.py

```diff
@@ -47,25 +49,34 @@ def create(self, request, *args, **kwargs):
         # Obtain the local storage path of the uploaded image
         image_path = serializer.instance.image.path
 
-        # Perform prediction
-        prediction_result = predictBloodCancer(image_path, self.model)
+        if self.model:
+            success = True
+            message = "The prediction was successful"
+            # Perform prediction
+            prediction_result = predictBloodCancer(image_path, self.model)
 
-        # Get the instance of the created model
-        instance = serializer.instance
+            # Get the instance of the created model
+            instance = serializer.instance
 
-        if prediction_result['is_cancer']:
-            instance.description = 'The uploaded image contains Blood cancer (Leukemia).'
-        else:
-            instance.description = 'The uploaded image does not contain cancer.'
-        instance.save()
+            if prediction_result['is_cancer']:
+                instance.description = 'The given sample is cancerous (leukemia).'
+            else:
+                instance.description = 'The given sample is non-cancerous.'
+            instance.save()
 
-        # Set the prediction_result field of the instance
-        instance.prediction_result = prediction_result
-        instance.save()
+            # Set the prediction_result field of the instance
+            instance.prediction_result = prediction_result
+            instance.save()
+        else:
+            success = False
+            message = "Unkown Error Occured, Please Try again later"
+            prediction_result = {}
+            
 
         # Customize the response data
         response_data = {
```

Comment:
In the response, success will serve as indicator boolean to the front-end whether the prediction was successful or not
message gives additional information
prediction returns the prediction result if it was successful,  otherwise an empty json